rest_framework_serializer_extensions/fields.py

```diff
@@ -59,15 +50,21 @@ class HashIdField(GetHashIdModelMixin, Field):
     Requires the source of the field to be an internal ID, and to provide
     a "model" keyword argument. Together these will produce the external ID.
     """
+    default_error_messages = {
+        'malformed_hash_id': 'That is not a valid HashId',
+    }
+
     def to_representation(self, value):
         return utils.external_id_from_model_and_internal_id(
             self.get_model(), value
         )
 
     def to_internal_value(self, value):
-        return utils.internal_id_from_model_and_external_id(
-            self.get_model(), value
-        )
+        model = self.get_model()
+        try:
+            return utils.internal_id_from_model_and_external_id(model, value)
```

Comment:
This isn't *necessarily* true from what I can see. I.e. it could be a perfectly valid HashID, but just not correspond to the model we're expecting?